src/Components/ChampionInfoComp/ChampionInfoComp.js

```diff
@@ -13,7 +13,7 @@ class ChampionInfoComp extends React.Component {
         if (this.props.champion) {  // champion was chosen
             if (!this.state.loadedPost || (this.state.loadedPost && this.state.loadedPost.title !== this.props.champion)) { // prevent infinite loops
```

Comment:
That's a weird comment. We shouldn't use hacks to prevent infinite loops. Let's try to understand why it is happening and solve the underlying problem.